File: VB/TimeEditControl/TimeEditPopupControl/CellViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Drawing

Namespace TimeEditControl

	Public Class CellViewInfo
		Public Enum CellType
			Hour
			Minute
			Format
		End Enum
		Private privateisHotPointed As Boolean
		Public Property isHotPointed() As Boolean
			Get
				Return privateisHotPointed
			End Get
			Set(ByVal value As Boolean)
				privateisHotPointed = value
			End Set
		End Property
		Private _isSelected As Boolean
		Public Property isSelected() As Boolean
			Get
				Return _isSelected
			End Get
			Set(ByVal value As Boolean)
				isHotPointed = False
				_isSelected = value
			End Set
		End Property
		Private privateType As CellType
		Public Property Type() As CellType
			Get
				Return privateType
			End Get
			Set(ByVal value As CellType)
				privateType = value
			End Set
		End Property
		Private privatecellTextFont As Font
		Public Property cellTextFont() As Font
			Get
				Return privatecellTextFont
			End Get
			Set(ByVal value As Font)
				privatecellTextFont = value
			End Set
		End Property
		Private privateCellText As String
		Public Property CellText() As String
			Get
				Return privateCellText
			End Get
			Set(ByVal value As String)
				privateCellText = value
			End Set
		End Property

		Public cellBounds As Rectangle = Rectangle.Empty

		Public Sub New(ByVal _cellTextFont As Font, ByVal _CellText As String, ByVal rectangle As Rectangle, ByVal enumType As CellType)
			cellTextFont = _cellTextFont
			CellText = _CellText
			cellBounds = rectangle
			Type = enumType
		End Sub
		Public Sub New()
		End Sub

		Protected Overrides Sub Finalize()
			Clear()
		End Sub

		Private Sub Clear()
			isHotPointed = False
			isSelected = False
			cellTextFont = Nothing
			cellBounds = Rectangle.Empty
			CellText = Nothing
		End Sub
	End Class

End Namespace

File: CS/TimeEditControl/TimeEditPopupControl/CellViewInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Drawing;

namespace TimeEditControl
{

    public class CellViewInfo
    {
        public enum CellType { Hour, Minute, Format };
        public bool isHotPointed { get; set; }
        private bool _isSelected;
        public bool isSelected
        {
            get
            {
                return _isSelected;
            }
            set
            {
                isHotPointed = false;
                _isSelected = value;
            }
        }
        public CellType Type { get; set; }
        public Font cellTextFont { get; set; }
        public string CellText { get; set; }

        public Rectangle cellBounds = Rectangle.Empty;

        public CellViewInfo(Font _cellTextFont, string _CellText, Rectangle rectangle, CellType enumType)
        {
            cellTextFont = _cellTextFont;
            CellText = _CellText;
            cellBounds = rectangle;
            Type = enumType;
        }
        public CellViewInfo() { }

        ~CellViewInfo()
        {
            Clear();
        }

        private void Clear()
        {
            isHotPointed = false;
            isSelected = false;
            cellTextFont = null;
            cellBounds = Rectangle.Empty;
            CellText = null;
        }
    }

}
